src/ansi-c/c_typecheck_code.cpp

```diff
@@ -243,7 +243,16 @@ void c_typecheck_baset::typecheck_decl(codet &code)
   }
 
   ansi_c_declarationt declaration;
+  irep_idt comment = code.source_location().get_comment();
   declaration.swap(code.op0());
+  if(!comment.empty())
+  {
+    for(auto &d : declaration.declarators())
+    {
+      if(d.source_location().get_comment().empty())
+        d.add_source_location().set_comment(comment);
```

Comment:
If the comment is non empty, should we merge instead of ignoring?

Reply:
The new implementation in #4666 also resolves this problem (in parts by just removing the above code :-) ).